src/wazuh_modules/agent_upgrade/manager/wm_agent_upgrade_commands.c

```diff
@@ -234,3 +252,77 @@ static void wm_agent_upgrade_start_upgrades(cJSON *json_response, const cJSON* t
         mtwarn(WM_AGENT_UPGRADE_LOGTAG, WM_UPGRADE_NO_AGENTS_TO_UPGRADE);
     }
 }
+
+char* wm_agent_upgrade_process_agent_result_command(const int* agent_ids, wm_upgrade_agent_status_task* task) {
+    cJSON *response = cJSON_CreateArray();
+    cJSON *message_array = cJSON_CreateArray();
+    // Only one id of agent will reach at a time
+    int agent_id = agent_ids[0];
+    cJSON *message_object = wm_agent_upgrade_parse_task_module_request(WM_UPGRADE_AGENT_STATUS, agent_id);
+    cJSON_AddStringToObject(message_object, "status", task->status);
```

Comment:
I think it is better to add `status` as a new parameter of `wm_agent_upgrade_parse_task_module_request`. If it is NULL, we don't add it to the JSON. If not, we add it.

Reply:
Agreed